src/graph/expression_operators.cpp

```diff
@@ -474,14 +474,16 @@ Expr dot(Expr a, Expr b, bool transA, bool transB, float scale) {
   // Currently only true when command line options
   // --optimize --cpu-thread=N with N > 0 are set.
   if(device == DeviceType::cpu) {
-    if(isFloat(aElementType) && (isFloat(bElementType) || isIntgemm(bElementType))) {
-      if(a->graph()->getBackend()->isInt8() || matchType<intgemm8>(bElementType)) {
+    if(isFloat(aElementType) && isFloat(bElementType)) {
+      return Expression<DotNodeOp>(clip(a, clipValue), clip(b, clipValue), transA, transB, scale);
+    } else if(isFloat(aElementType) && isIntgemm(bElementType)) {
+      // @TODO: this branch should move into cpu::integer::*
+      if(sizeOf(bElementType) == 1) {
         return cpu::integer::dot<Type::int8>(a, b, transA, transB, scale);
-      } else if(a->graph()->getBackend()->isInt16() || matchType<intgemm16>(bElementType)) {
+      } else if(sizeOf(bElementType) == 2) {
```

Comment:
Again, what does `sizeOf(bElementType) == 2)` and can we use `matchType` or something else more obvious instead?